src/core/reference/include/openvino/reference/fake_convert.hpp

```diff
@@ -0,0 +1,329 @@
+            /* round to nearest even, if rne_mask is enabled */
+            /* 0 00000 0010000000, find grs patterns */
+            // 0xx - do nothing
+            // 100 - this is a tie : round up if the mantissa's bit just before G is 1, else do nothing
+            // 101, 110, 111 - round up > 0x0080
+            h.u += (((rnmask > 0x0080) || (rnmask_tie == rne_tie)) << lshift);
+        }
+        h.u = (h.u & mask_mant); /* truncation */
+        out[i] = h.f;
+    }
+}
+
+/**
+ * @brief Emulation of conversion fp16 value to f8e4m3 format
+ *
+ * @param arg       Pointer to the input data.
+ * @param out       Pointer to the otuput data.
+ * @param count     Number of elements in the data input.
+ *
+ * Exponent denormal values 0 -7
+ * Exponent normal values 1..15 -6..8 (7 - exponent)
+ * Exponent NaN values 15 8
+ *
+ */
+template <typename T>
+void emulate_f8e4m3_on_fp16(const T* arg,
+                            T* out,
+                            size_t count,
+                            int exp_bits = 5,
+                            int mbits = 9,
+                            bool use_clamp = true) {
+    typedef union half_t {
+        unsigned short u;
+        T f;
+    } __half_t;
+
+    const auto non_mant_bits = exp_bits + 1; /* exponent + sign */        ///  6 - ?
+    const auto lshift = 10 - (mbits - non_mant_bits);                     /// 10 - (9 - 6) == 7 - ???
+    const unsigned short rne_mask = 1;                                    /* round to nearest even mask */
+    const unsigned short mask_mant = (unsigned short)(0xFFFF << lshift);  // 1111111111111111 -> 1 11111 1111000000
+    constexpr unsigned short grs_bitmask = 0x007F;                        /// 0 00000 0001111111
+    constexpr unsigned short rne_tie = 0x00C0;                            /// 0 00000 0011000000
+
+    __half_t h;
+    for (size_t i = 0; i < count; ++i) {
+        h.f = arg[i];
+        float inval = ngraph::float16(arg[i]);
+        /* flush values below 1-4-3 (offset=4) subnormal range to zero */
+        if (fabs(inval) < 0.001953125)  // 2**-9
+            h.f = 0;
+
+        short exp_h = (short)((h.u & 0x7C00) >> 10) -
+                      15;                   /// 0111110000000000 -> 0000000000011111 - 15 - biased exponent for fp16
+        short sign_h = (h.u & 0x8000);      /// & 1 00000 0000000000
+        short mantissa_h = (h.u & 0x03FF);  /// & 0 00000 1111111111
+        ///(h.u && 0111111111111111) < 0 10010 1110000000 (19326) - ????
+        unsigned short can_round = ((h.u & 0x7FFF) < 0b101111110000000) ? 1 : 0;
+        unsigned short is_naninf = ((h.u & 0x7C00) == 0x7C00) ? 1 : 0;
+
+        int dshift = 0;
+        if (exp_h > 8) {  // too large, set it to NaN or inf
+            is_naninf = 1;
+            if (use_clamp) {
+                exp_h = 8;
+                mantissa_h = 0b0000001100000000;
+            } else {
+                mantissa_h = 0;
+                exp_h = 16;
```

Comment:
```suggestion
                exp_h = 16;
```
Make defines for all magic values to know what they mean.

Reply:
I've created constexpr for the values with meaningful associations, for others leaving as is in the POC, can be refactored as a follow up.